rocketpy/mathutils/function.py

```diff
@@ -1144,6 +1144,49 @@ def low_pass_filter(self, alpha, file_path=None):
             title=self.title,
         )
 
+    def remove_outliers_iqr(self, threshold=1.5):
+        """Remove outliers from the Function source using the interquartile
+        range method.
+
+        Parameters
+        ----------
+        threshold : float, optional
+            Threshold for the interquartile range method. Default is 1.5.
+
+        Returns
+        -------
+        Function
+            The Function with the outliers removed.
+
+        References
+        ----------
+        [1] https://en.wikipedia.org/wiki/Outlier#Tukey's_fences
+        """
+
+        if callable(self.source):
+            print("Cannot remove outliers if the source is a callable object.")
+            return self
```

Comment:
My suggestion here would be to raise an exception, so that this method does not misled the user by returning normally.

Check out `set_discrete_based_on_model`, the `Function` class normally throws an error when the operation cannot be executed instead of returning a default.

Reply:
Solved here: [5987bd8](https://github.com/RocketPy-Team/RocketPy/pull/554/commits/5987bd8fb267daf95a6ed345849c3b218dcfb838)